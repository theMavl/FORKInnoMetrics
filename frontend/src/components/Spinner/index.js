import React from 'react'
import styles from './style.css'

class Spinner extends React.Component {
  render() {
    return (
        <div className={styles.spinnerContainer} {...this.props}>
          <div className={styles.spinner}>
            <div className={styles.sc1}></div>
            <div className={styles.sc2}></div>
            <div className={styles.sc3}></div>
            <div className={styles.sc4}></div>
            <div className={styles.sc5}></div>
            <div className={styles.sc6}></div>
            <div className={styles.sc7}></div>
            <div className={styles.sc8}></div>
          </div>
        </div>

    )
  }
}

export default Spinner