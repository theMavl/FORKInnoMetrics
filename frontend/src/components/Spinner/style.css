.spinnerContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

.spinner {
    position: relative;
    width: 125px;
    height: 125px;
    margin: auto;
    transform: scale(0.6);
}

.smallCircle{
    position:absolute;
    background-color:rgb(0, 0, 0);
    height:22px;
    width:22px;
    border-radius:12px;
    animation-name:fade;
    animation-duration:1.2s;
    animation-iteration-count:infinite;
    animation-direction:normal;
}

.sc1{
    composes: smallCircle;
    left:0;
    top:51px;
    animation-delay:0.45s;
}

.sc2{
    composes: smallCircle;
    left:15px;
    top:15px;
    animation-delay:0.6s;
}

.sc3{
    composes: smallCircle;
    left:51px;
    top:0;
    animation-delay:0.75s;
}

.sc4{
    composes: smallCircle;
    right:15px;
    top:15px;
    animation-delay:0.9s;
}

.sc5{
    composes: smallCircle;
    right:0;
    top:51px;
    animation-delay:1.05s;
}

.sc6{
    composes: smallCircle;
    right:15px;
    bottom:15px;
    animation-delay:1.2s;
}

.sc7{
    composes: smallCircle;
    left:51px;
    bottom:0;
    animation-delay:1.35s;
}

.sc8{
    composes: smallCircle;
    left:15px;
    bottom:15px;
    animation-delay:1.5s;
}



@keyframes fade{
    0%{
        background-color:rgb(0,0,0);
    }

    100%{
        background-color:rgb(255,255,255);
    }
}