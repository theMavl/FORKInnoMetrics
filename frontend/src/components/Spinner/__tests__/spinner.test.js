import React from 'react'
import TestRenderer from 'react-test-renderer'
import Spinner from '../index'

describe('Test Spinner component', () => {

  it('Test render Spinner component', () => {

    const instance = TestRenderer.create(
      <Spinner/>
    )
    expect(instance.toJSON()).toMatchSnapshot()
  })
})