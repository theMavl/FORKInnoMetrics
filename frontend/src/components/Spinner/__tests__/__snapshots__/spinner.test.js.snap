// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Spinner component Test render Spinner component 1`] = `
<div>
  <div>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
